internal/clients/client_models.go

```diff
@@ -30,4 +30,9 @@ type InstanceType struct {
 
 	// The size of the memory, in MiB.
 	Memory int64 `json:"memory,omitempty"`
+
+	// Does the instance type supports RHEL
+	Supported bool `json:"supported,omitempty"`
+
+	SupportedArchitectures []string `json:"supported_architectures,omitempty"`
```

Comment:
As we discussed on Slack, let's break multiple architectures into multiple InstanceType values each having it's own architecture.

I suggest that you create constant for each architecture in this package and validate them (e.g. in switch-case with a default branch with error), because different clouds might have different architecture names (e.g. x86-64 vs x86_64). We should only use one, pick something and insist on it (convert), return errors on unknown architectures (we need to manually add support for new ones).